kale/predict/losses.py

```diff
@@ -274,3 +275,80 @@ def _moment_k(x: torch.Tensor, domain_labels: torch.Tensor, k_order=2):
             moment_sum += euclidean(x_k_order[i], x_k_order[j])
             n_pair += 1
     return moment_sum / n_pair
+
+
+class protonet_loss:
+    """
+    ProtoNet loss function.
+    This is a loss function for prototypical networks and N-way-K-shot problems. It computes the loss and
+    accuracy of the model by measuring the Euclidean distance between features of support samples and query samples.
+
+    Because the loss requests some constant hyperparameters, it is not a general function but defined as a class.
+
+    Args:
+        num_classes (int): Number of classes in a task
+        num_query_samples (int): Number of query samples per class
+        device (torch.device): The desired device of returned tensor
+
+    Examples:
+        >>> loss_fn = protonet_loss(num_classes=5, num_query_samples=15, device=torch.device("cuda"))
+        >>> loss, acc = loss_fn(feature_support, feature_query)
+
+
+    Reference:
+        Snell, J., Swersky, K. and Zemel, R., 2017. Prototypical Networks for Few-shot Learning.
+        Advances in Neural Information Processing Systems, 30.
```

Comment:
Good here. So you know about it, just not checking everything before requesting for review.